src/persist/src/indexed/runtime.rs

```diff
@@ -735,13 +903,73 @@ impl<K: Codec, V: Codec> Iterator for DecodedSnapshotIter<K, V> {
     }
 }
 
+#[derive(Debug, Clone)]
+enum CmdReadSender {
+    Full(crossbeam_channel::Sender<RuntimeCmd>),
+    Read(crossbeam_channel::Sender<RuntimeReadCmd>),
+}
+
+impl CmdReadSender {
+    fn send(&self, cmd: CmdRead) {
+        match self {
+            CmdReadSender::Full(tx) => {
+                if let Err(crossbeam_channel::SendError(cmd)) =
+                    tx.send(RuntimeCmd::Cmd(Cmd::CmdRead(cmd)))
+                {
+                    // According to the docs, a SendError can only happen if the
+                    // receiver has hung up, which in this case only happens if the
+                    // thread has exited. The thread only exits if we send it a
+                    // Cmd::Stop (or it panics).
+                    fill_runtime_cmd_res_runtime_shutdown(cmd)
+                }
+            }
+            CmdReadSender::Read(tx) => {
+                if let Err(crossbeam_channel::SendError(cmd)) =
+                    tx.send(RuntimeReadCmd::CmdRead(cmd))
+                {
+                    // According to the docs, a SendError can only happen if the
+                    // receiver has hung up, which in this case only happens if the
+                    // thread has exited. The thread only exits if we send it a
+                    // Cmd::Stop (or it panics).
+                    fill_runtime_read_cmd_res_runtime_shutdown(cmd)
+                }
+            }
+        }
+    }
+
+    fn stop(&self, tx: PFutureHandle<()>) {
```

Comment:
I don't like this one, it's nearly an exact duplicate of `send`, but don't see anything nicer to do here. one option I suppose is to push Stop/Close into CmdRead and make Indexed handle it, but then the RuntimeImpl loops will have to sniff this out so it can flush out `step`, which is unsatisfying. I'm not convinced that there isn't something obviously much nicer to do here